ql/src/test/results/clientpositive/llap/excluded_rule_explain.q.out

```diff
@@ -0,0 +1,112 @@
+PREHOOK: query: EXPLAIN CBO SELECT 1
+PREHOOK: type: QUERY
+PREHOOK: Input: _dummy_database@_dummy_table
+#### A masked pattern was here ####
+POSTHOOK: query: EXPLAIN CBO SELECT 1
+POSTHOOK: type: QUERY
+POSTHOOK: Input: _dummy_database@_dummy_table
+#### A masked pattern was here ####
+Excluded rules: HiveJoinPushTransitivePredicatesRule|HivePreFilteringRule
+
+CBO PLAN:
+HiveProject(_o__c0=[1])
+  HiveTableScan(table=[[_dummy_database, _dummy_table]], table:alias=[_dummy_table])
+
+PREHOOK: query: EXPLAIN FORMATTED CBO SELECT 1
+PREHOOK: type: QUERY
+PREHOOK: Input: _dummy_database@_dummy_table
+#### A masked pattern was here ####
+POSTHOOK: query: EXPLAIN FORMATTED CBO SELECT 1
+POSTHOOK: type: QUERY
+POSTHOOK: Input: _dummy_database@_dummy_table
+#### A masked pattern was here ####
+{"CBOPlan":"{\n  \"rels\": [\n    {\n      \"id\": \"0\",\n      \"relOp\": \"org.apache.hadoop.hive.ql.optimizer.calcite.reloperators.HiveTableScan\",\n      \"table\": [\n        \"_dummy_database\",\n        \"_dummy_table\"\n      ],\n      \"table:alias\": \"_dummy_table\",\n      \"inputs\": [],\n      \"rowCount\": 1.0,\n      \"avgRowSize\": 129.0,\n      \"rowType\": [\n        {\n          \"type\": \"BIGINT\",\n          \"nullable\": true,\n          \"name\": \"BLOCK__OFFSET__INSIDE__FILE\"\n        },\n        {\n          \"type\": \"VARCHAR\",\n          \"nullable\": true,\n          \"precision\": 2147483647,\n          \"name\": \"INPUT__FILE__NAME\"\n        },\n        {\n          \"fields\": [\n            {\n              \"type\": \"BIGINT\",\n              \"nullable\": true,\n              \"name\": \"writeid\"\n            },\n            {\n              \"type\": \"INTEGER\",\n              \"nullable\": true,\n              \"name\": \"bucketid\"\n            },\n            {\n              \"type\": \"BIGINT\",\n              \"nullable\": true,\n              \"name\": \"rowid\"\n            }\n          ],\n          \"name\": \"ROW__ID\"\n        },\n        {\n          \"type\": \"BOOLEAN\",\n          \"nullable\": true,\n          \"name\": \"ROW__IS__DELETED\"\n        }\n      ]\n    },\n    {\n      \"id\": \"1\",\n      \"relOp\": \"org.apache.hadoop.hive.ql.optimizer.calcite.reloperators.HiveProject\",\n      \"fields\": [\n        \"_o__c0\"\n      ],\n      \"exprs\": [\n        {\n          \"literal\": 1,\n          \"type\": {\n            \"type\": \"INTEGER\",\n            \"nullable\": false\n          }\n        }\n      ],\n      \"rowCount\": 1.0\n    }\n  ]\n}","cboInfo":"Excluded rules: HiveJoinPushTransitivePredicatesRule|HivePreFilteringRule"}
+PREHOOK: query: EXPLAIN SELECT 1
+PREHOOK: type: QUERY
+PREHOOK: Input: _dummy_database@_dummy_table
+#### A masked pattern was here ####
+POSTHOOK: query: EXPLAIN SELECT 1
+POSTHOOK: type: QUERY
+POSTHOOK: Input: _dummy_database@_dummy_table
+#### A masked pattern was here ####
+Excluded rules: HiveJoinPushTransitivePredicatesRule|HivePreFilteringRule
```

Comment:
That's not displayed for `EXPLAIN CBO` (see https://github.com/apache/hive/blob/2b7c3a844156077a8a57f75b548aeb7bc7685cfe/ql/src/java/org/apache/hadoop/hive/ql/parse/ExplainSemanticAnalyzer.java#L204).

Reply:
This query is not an `EXPLAIN CBO`. Am I missing something?

Reply:
Sorry, I must have missed the line you were mentioning. In this specific case it's the `hive.explain.user` property that is missing. The default value is `true`, however, there must be an override somewhere because it's not activated by default.

When setting `set hive.explain.user=true;`, `EXPLAIN SELECT 1;` prints `Plan optimized by CBO. Excluded rules: HiveJoinPushTransitivePredicatesRule|HivePreFilteringRule` instead of the simple `Excluded rules: HiveJoinPushTransitivePredicatesRule|HivePreFilteringRule`.

If you want I can add what follows at the end of the .q file:
```
set hive.explain.user=true;
EXPLAIN SELECT 1;
```

WDYT?

Reply:
Thanks for the clarification. No need to add more tests, I just wanted to make sure we didn't break something by accident.